web/src/types/storage/model.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) [2025] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import * as apiModel from "~/api/storage/types/config-model";
+
+type Drive = apiModel.Drive;
+
+type LogicalVolume = apiModel.LogicalVolume;
+
+interface VolumeGroup extends Omit<apiModel.VolumeGroup, "targetDevices" | "logicalVolumes"> {
```

Comment:
Why interface?

Reply:
I see. I used to do something like

```ts
type VolumeGroup = Omit<apiModel.VolumeGroup, "targetDevices" | "logicalVolumes">  & {
  targetDevices: Drive[];
  logicalVolumes: LogicalVolume[];
}
```

Reply:
Do we want to use type intersections for extending a type? AFAIS, extending interfaces is preferred:

Reply:
JFTR: This article is also interesting https://blog.logrocket.com/types-vs-interfaces-typescript/.

About the video, it defends type over interface to avoid the interface declaration merging and to be able to use unions. IMHO, the discussion is not about using either interfaces or types, but when to use one or the other.

For exteding types, all the resources I have read recommend interfaces. It seems that React also uses the very same approach as the code proposed in this PR (i.e., extend on Omit). Of course, type alias makes a lot of sense for other use cases like unions, etc. IMHO, I would use interface when we need to extend a type and type alias for the rest.

Reply:
The video was just a summary, but fine with me to use both if there are reasons to do so.

Reply:
JFTR, due the little advantage of using both I was most all for using just one and keep things simple, but don't want to invest too much time in _code style_ discussions since they are not too much useful right now.

Reply:
As discussed in a meeting, we found useful to use interface for extending types. We will add a new section in the development resources of the agama documentation, https://agama-project.github.io/docs/devel